src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -322,10 +343,31 @@ private static boolean branchContainsJavadocComment(DetailAST token) {
             }
             curNode = toVisit;
         }
-
         return result;
     }
 
+    /**
+     * Checks whether a javadoc follows the specified comment pattern.
+     *
+     * @param token tree token
+     * @return true if the javadoc comment of the specified token follows the given pattern
+     */
+    private boolean isValidJavadocComment(DetailAST token) {
+        final Pattern javadocStars = Pattern.compile("\\*\\s*");
+        final Pattern javadocWhiteSpaces = Pattern.compile("\\s+");
```

Comment:
please do pattern as final field, we need to compile them only ones, not on each method execution.